### PR TITLE
[FSSDK-9775] Log invalid user attributes

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary

- Log discovered invalid user attributes to an optional logger when BuildAttributeList

## Test plan

- Existing tests should pass
- Add new test coverage

## Issues
- FSSDK-9775

### CODE REVIEW

File: OptimizelySDK/Event/EventFactory.cs [hunk 1]
```diff
@@ -192,19 +192,32 @@ ProjectConfig config
...
-                foreach (var validUserAttribute in userAttributes.Where(attribute =>
-                             Validator.IsUserAttributeValid(attribute)))
+                foreach (var userAttribute in userAttributes)
                 {
-                    var attributeId = config.GetAttributeId(validUserAttribute.Key);
-                    if (!string.IsNullOrEmpty(attributeId))
+                    if (Validator.IsUserAttributeValid(userAttribute))
+                    {
+                        var attributeId = config.GetAttributeId(userAttribute.Key);
+                        if (!string.IsNullOrEmpty(attributeId))
```
Comment (jaeopt) [inline OptimizelySDK/Event/EventFactory.cs:204]: What about switching 204 and 201, so we only log invalid type attributes for attribute ids defined in datafile.
We do not care about types of not-registered attributes.

**Review by jaeopt — CHANGES_REQUESTED** [review]
A suggestion. We can discuss about the options.

File: OptimizelySDK/Event/EventFactory.cs (hunk 1 above)
Comment (mikechu-optimizely) [reply OptimizelySDK/Event/EventFactory.cs]: Let me know what you think @jaeopt

**Comment by mikechu-optimizely** [comment]
Looks like I'm going to need to work on the C# test app to pass FSC

File: OptimizelySDK/Event/EventFactory.cs [hunk 2]
```diff
@@ -192,18 +192,40 @@ ProjectConfig config
...
                     }
+
+                    attributesList.Add(new VisitorAttribute(attributeId, userAttribute.Key,
+                        CUSTOM_ATTRIBUTE_FEATURE_TYPE, userAttribute.Value));
+                }
+
+                if (notRegisteredAttributeKeys.Count > 0 && !(logger is null))
+                {
+                    logger.Log(LogLevel.WARN,
+                        $"User attributes: {string.Join(", ", notRegisteredAttributeKeys.ToArray())} are not supported by the datafile and will not be used.");
```
Comment (jaeopt) [inline OptimizelySDK/Event/EventFactory.cs:222]: Not sure if we want to log all these here. I see some clients have 100+ attributes not all registered to us. Each event will dump a bunch of these messages. Let's discuss.

**Review by jaeopt — CHANGES_REQUESTED** [review]
The changes look good! 
One clarifying question about if we want to log all non-registered attributes or not.

**Comment by mikechu-optimizely** [comment]
This was a spike only Jira ticket.  Woops. 🤷

**Pull request closed without merging**